include/deal.II/lac/vector_type_traits.h

```diff
@@ -0,0 +1,46 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2017 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE at
+// the top level of the deal.II distribution.
+//
+// ---------------------------------------------------------------------
+
+#ifndef dealii__vector_type_traits_h
+#define dealii__vector_type_traits_h
+
+#include <deal.II/base/std_cxx11/type_traits.h>
+
+
+DEAL_II_NAMESPACE_OPEN
+
+
+/**
+ * Type trait for a serial vector, i.e. a vector class which storage is not
+ * distributed over processes.
+ * The specialization
+ *   struct is_serial_vector : std_cxx11::true_type,
+ * respectively,
+ *   struct is_serial_vector : std_cxx11::false_type
+ * for a vector type with storage distributed over processes,
+ * must be done in a header file of a vector declaration.
+ *
+ * @author Uwe Köcher, 2017
+ */
+template<typename T>
+// struct is_serial_vector : std_cxx11::false_type
+struct is_serial_vector
+{
+};
```

Comment:
I think the better approach would be to just not declare the type at all -- i.e., to just say that a template exists
```
  template <typename T>
  struct is_serial_vector;
```
but not give it a body. That way, if a piece of ever tried to use the general template for any value of `T` for which no specialization of `is_serial_vector` is available, the compiler will say "incomplete type", rather than "no member `abc` found in class `is_serial_vector<...>`".